<launch>
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<arg name="camera"  default="camera" />

	<group ns="grasping_teleop_key">
		<node name="grasping_teleop_key" pkg="sigverse_turtlebot3_open_manipulator" type="grasping_teleop_key" output="screen" launch-prefix="gnome-terminal -t 'TurtleBot3 teleop key' -- ">
		<!--
		<remap from="/tb3omc/joint_state"         to="/tb3omc/joint_state" />
		<remap from="/tb3omc/cmd_vel"             to="/tb3omc/cmd_vel" />
		<remap from="/tb3omc/joint_trajectory"    to="/tb3omc/joint_trajectory" />
		-->
		</node>
	</group>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>

	<node pkg="image_view" type="image_view" name="image_view" args="image:=$(arg camera)/rgb/image_raw"/>

</launch>

